### ISSUE TITLE
Duplicate status messages

### ISSUE THREAD

**Issue by technoir42** [Flank/flank#841]
**Describe the bug**
Same status messages are printed multiple times:
```
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-1 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-4 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-0 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-3 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-2 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-1 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-4 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-0 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-3 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-2 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-1 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-4 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-0 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-3 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-2 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-1 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-4 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-0 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-3 PENDING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-2 PENDING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-1 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-4 RUNNING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-0 PENDING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-3 RUNNING
  0m  0s matrix-12gn3h80wl2ll Pixel2-27 shard-2 PENDING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-1 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-4 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-0 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-3 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-2 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-1 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-4 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-0 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-3 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-2 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-1 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-4 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-0 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-3 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-2 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-1 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-4 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-0 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-3 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-2 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-1 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-4 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-0 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-3 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-2 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-1 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-4 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-0 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-3 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-2 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-1 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-4 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-0 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-3 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-2 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-1 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-4 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-0 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-3 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-2 RUNNING
  1m 15s matrix-12gn3h80wl2ll Pixel2-27 shard-1 Preparing device.
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-4 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-0 RUNNING
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-3 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-2 RUNNING
  1m 15s matrix-12gn3h80wl2ll Pixel2-27 shard-1 Preparing device.
  0m 23s matrix-12gn3h80wl2ll Pixel2-27 shard-4 RUNNING
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-0 RUNNING
  1m 21s matrix-12gn3h80wl2ll Pixel2-27 shard-3 Preparing device.
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-2 RUNNING
  1m 15s matrix-12gn3h80wl2ll Pixel2-27 shard-1 Preparing device.
  1m 26s matrix-12gn3h80wl2ll Pixel2-27 shard-4 Preparing device.
  1m 26s matrix-12gn3h80wl2ll Pixel2-27 shard-0 Preparing device.
  1m 21s matrix-12gn3h80wl2ll Pixel2-27 shard-3 Preparing device.
  0m 29s matrix-12gn3h80wl2ll Pixel2-27 shard-2 RUNNING
  1m 15s matrix-12gn3h80wl2ll Pixel2-27 shard-1 Preparing device.
  1m 26s matrix-12gn3h80wl2ll Pixel2-27 shard-4 Preparing device.
  1m 26s matrix-12gn3h80wl2ll Pixel2-27 shard-0 Preparing device.
  1m 21s matrix-12gn3h80wl2ll Pixel2-27 shard-3 Preparing device.
  1m 32s matrix-12gn3h80wl2ll Pixel2-27 shard-2 Preparing device.
  1m 15s matrix-12gn3h80wl2ll Pixel2-27 shard-1 Preparing device.
  1m 26s matrix-12gn3h80wl2ll Pixel2-27 shard-4 Preparing device.
  1m 26s matrix-12gn3h80wl2ll Pixel2-27 shard-0 Preparing device.
  1m 21s matrix-12gn3h80wl2ll Pixel2-27 shard-3 Preparing device.
  1m 32s matrix-12gn3h80wl2ll Pixel2-27 shard-2 Preparing device.
  1m 15s matrix-12gn3h80wl2ll Pixel2-27 shard-1 Preparing device.
  1m 26s matrix-12gn3h80wl2ll Pixel2-27 shard-4 Preparing device.
  1m 26s matrix-12gn3h80wl2ll Pixel2-27 shard-0 Preparing device.
  1m 21s matrix-12gn3h80wl2ll Pixel2-27 shard-3 Preparing device.
  1m 32s matrix-12gn3h80wl2ll Pixel2-27 shard-2 Preparing device.
  1m 15s matrix-12gn3h80wl2ll Pixel2-27 shard-1 Preparing device.
  1m 26s matrix-12gn3h80wl2ll Pixel2-27 shard-4 Preparing device.
  1m 26s matrix-12gn3h80wl2ll Pixel2-27 shard-0 Preparing device.
  1m 21s matrix-12gn3h80wl2ll Pixel2-27 shard-3 Preparing device.
  1m 32s matrix-12gn3h80wl2ll Pixel2-27 shard-2 Preparing device.
  1m 55s matrix-12gn3h80wl2ll Pixel2-27 shard-1 Installing apps.
  1m 26s matrix-12gn3h80wl2ll Pixel2-27 shard-4 Preparing device.
  1m 26s matrix-12gn3h80wl2ll Pixel2-27 shard-0 Preparing device.
  1m 55s matrix-12gn3h80wl2ll Pixel2-27 shard-3 Installing apps.
  1m 32s matrix-12gn3h80wl2ll Pixel2-27 shard-2 Preparing device.
  1m 55s matrix-12gn3h80wl2ll Pixel2-27 shard-1 Installing apps.
  2m  1s matrix-12gn3h80wl2ll Pixel2-27 shard-4 Installing apps.
  2m  1s matrix-12gn3h80wl2ll Pixel2-27 shard-0 Installing apps.
  1m 55s matrix-12gn3h80wl2ll Pixel2-27 shard-3 Installing apps.
  1m 32s matrix-12gn3h80wl2ll Pixel2-27 shard-2 Preparing device.
```

**To Reproduce**
Run Flank v20.06.0 using Fladle.

**Expected behavior**
Each status message should be unique.

**Details:**
```
version: v20.06.0
revision: 77f18db6109b49496fe82c99a553f16414f7366a
```
Fladle v0.10.1

**Comment by bootstraponline** [comment]
Thank you for reporting this bug!

**Comment by jan-gogo** [comment]
Hi @technoir42 try to run flank with option `--output-style=single`, your console may doesn't support all ansi codes. Actually flank is providing 3 ways to display test executions status. 

https://github.com/Flank/flank/blob/23d66ed9d4a384331ac1cb1910be88f8df7241e4/test_runner/flank.yml#L207
> Output style of execution status. May be one of [verbose, multi, single]. 
For runs with only one test execution the default value is `verbose`, in other cases `multi` is used as the default. The output style 'multi' is not displayed correctly on consoles which don't support ansi codes, to avoid corrupted output use `single` or `verbose`.

**Comment by technoir42** [comment]
@jan-gogo I'm using Fladle. Did the default value for `output-style` change between 20.05.x and 20.06.x? I don't have this issue with 20.05.2 using the same Fladle settings.

**Comment by bootstraponline** [comment]
@technoir42 where are you running flank? Circle CI?

**Comment by technoir42** [comment]
@bootstraponline the output I posted is from the terminal when I ran tests from my local machine.

**Comment by jan-gogo** [comment]
@bootstraponline I will change default output for multiple executions to `single`.

**Comment by bootstraponline** [comment]
Is there a way for us to detect what the terminal supports and adjust accordingly? I'm using iTerm and Flank works as expected.

**Comment by bootstraponline** [comment]
I think default to single makes sense.

**Comment by jan-gogo** [comment]
> Is there a way for us to detect what the terminal supports and adjust accordingly?

First check on google gives me:
https://stackoverflow.com/questions/7445658/how-to-detect-if-the-console-does-support-ansi-escape-codes-in-python
It is possible in python so probably in jvm too

**Comment by technoir42** [comment]
I tried running flank directly without Fladle and `--output-style=multi` works correctly in iTerm, that is for 4 shards there are only 2 status lines:
```
  2m 25s matrix-3aj1tsex29nuo Pixel2-27 shard-3 Starting instrumentation test.
  2m 25s matrix-3aj1tsex29nuo Pixel2-27 shard-0 Starting instrumentation test.
  2m 36s matrix-3aj1tsex29nuo Pixel2-27 shard-2 Starting instrumentation test.
  2m 25s matrix-3aj1tsex29nuo Pixel2-27 shard-1 Starting instrumentation test.
```
This actually looks cool.
Perhaps Fladle should pass `--output-style=single`? Or could there be a bug piping the output from Flank through Gradle?

**Comment by bootstraponline** [comment]
yeah this seems like a Fladle bug and not a Flank bug.

/cc @runningcode

**Comment by jan-gogo** [comment]
> piping the output from Flank through Gradle

I vote that could be source of this problem. Running flank through Intellij with Gradle gives the same issue .

**Comment by runningcode** [comment]
Thanks for bringing this to my attention. Could you please file a bug against fladle? If you could indicate a way for me to test that the fix works (like a small repro) that would be very helpful. What does `--output-style=multi` vs `--output-style=single` do?
Is there ever a use case for `multi` in fladle? Should it be a configuration option?

**Comment by jan-gogo** [comment]
> Could you please file a bug against fladle

@runningcode https://github.com/runningcode/fladle/issues/140

**Comment by runningcode** [comment]
This is fixed in fladle 0.10.2 https://runningcode.github.io/fladle/changelog/#0102
